scapy/layers/ppp.py

```diff
@@ -552,18 +552,163 @@ class PPP_LCP_Discard_Request(PPP_LCP):
                    IntField("magic_number", None),
                    StrLenField("data", "", length_from=lambda p:p.len-8)]
 
+### Password authentication protocol (RFC 1334)
+
+_PPP_paptypes = {1: "Authenticate-Request",
+                 2: "Authenticate-Ack",
+                 3: "Authenticate-Nak"}
+
+
+class PPP_PAP(Packet):
+    name = "PPP Password Authentication Protocol"
+    fields_desc = [ByteEnumField("code", 1, _PPP_paptypes),
+                   XByteField("id",0),
+                   ShortField("len", None),
```

Comment:
This should probably be a `FieldLenField("len", None, length_of="data", fmt="!H")`. Then you don't need your specific `post_build()`.